### ISSUE TITLE
mkbundle: ' error: ‘mono_aot_register_module’ undeclared (first use in this function)'

### ISSUE THREAD

**Issue by elitest** [mono/mono#9956]
## Steps to Reproduce

1. Install mono nightlies on ubuntu 16.04 according to mono docs using mono repos
2. Download IronPython latest binary release
3. Run 'mkbundle -o IPY -L /usr/lib/mono/4.5 ipy.exe'

### Current Behavior

```
temp.c: In function ‘init_default_mono_api_struct’:
temp.c:116:33: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
  mono_api.mono_jit_set_aot_mode = mono_jit_set_aot_mode;
                                 ^
temp.c:117:38: error: ‘mono_aot_register_module’ undeclared (first use in this function)
  mono_api.mono_aot_register_module = mono_aot_register_module;
                                      ^
temp.c:117:38: note: each undeclared identifier is reported only once for each function it appears in
temp.c: At top level:
temp.c:92:1: warning: ‘validate_api_struct’ defined but not used [-Wunused-function]
 validate_api_struct ()
 ^
temp.c:111:1: warning: ‘init_default_mono_api_struct’ defined but not used [-Wunused-function]
 init_default_mono_api_struct ()
 ^
ERROR: [Fail] 256
```

### Expected Behavior
It should bundle the application right?

## On which platforms did you notice this

[ ] macOS
[X ] Linux
[ ] Windows

**Version Used**:
```
Mono JIT compiler version 5.17.0.535 (tarball Mon Aug  6 10:01:05 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(4)
	Suspend:       preemptive
	GC:            sgen (concurrent by default)
```

## Stacktrace

See above

I assume this is related to https://github.com/mono/mono/blob/170e9442c976ed6cd4a499093e11a9e12e7368bb/mcs/tools/mkbundle/mkbundle.cs#1049.  I'm still learning C# so let me know if you have any suggestions for how to get this to a PR.

**Comment by petlof** [comment]
I'm having the same issue here when upgrading to Mono 5.14

Mono JIT compiler version 5.14.0.177 (tarball Sat Aug  4 08:28:13 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          yes(3.6.0svn-mono-/)
        GC:            sgen (concurrent by default)

**Comment by elitest** [comment]
So I've had some time to dig into this further.  Reading the error message more closely it is actually a C error from when it tries to compile the intermediate native C code that wraps the .Net bits.
/mcs/tools/mkbundle/bundle-mono-api.inc and /mcs/tools/mkbundle/template_common.inc are the basis of what the resulting file of the source is built from.  Line [57](https://github.com/mono/mono/blob/170e9442c976ed6cd4a499093e11a9e12e7368bb/mcs/tools/mkbundle/template_common.inc#L57) matches up with the error from the resulting C source file I have.

This looks like it was part of some refactoring of mkbundle done by @grendello a few months ago.  Can you comment to shed some light on this perhaps?

**Comment by alexanderkyte** [comment]
This function in question belongs to the mono support for working with AOT'ed code. It sounds like the C stub for the generated C isn't including the header that exports this function.

**Comment by elitest** [comment]
Thanks @alexanderkyte.  Indeed [mono/mini/aot-runtime.h](https://github.com/mono/mono/blob/3dd1e1d92abd6c5586dfacc81003dc5713800847/mono/mini/aot-runtime.h) looks to be the culprit.  If I tell mkbundle to generate a intermediate C filem add "#include <mono/mini/aot-runtime.h>" and then try to compile using:
```
gcc -o ipy ipy.c `pkg-config --cflags --libs mono-2`
```
It tells me it can't find the header:
```
ipy.c:3:35: fatal error: mono/mini/aot-runtime.h: No such file or directory
compilation terminated.
```
looking in /usr/include/mono-2.0 it doesn't seem like 'mini' exists, and aot-runtime.h isn't included in any of the other folders.
```
ls /usr/include/mono-2.0/mono/
cil  jit  metadata  utils
```

If I download a copy of the master branch and include all the folders I end up in dependency hell:
```
gcc -I/home/elitest/Dev/mono-master -o ipy ipy.c `pkg-config --cflags --libs mono-2`
In file included from /home/elitest/Dev/mono-master/mono/mini/aot-runtime.h:11:0,
                 from ipy.c:3:
/home/elitest/Dev/mono-master/mono/mini/mini.h:11:20: fatal error: config.h: No such file or directory
```
Is there something wrong with my setup? Or should is there another location I should be including?  Or is this a another bug?

**Comment by alexanderkyte** [comment]
The linked PR should clarify things a bit. The AOT runtime support requires access to private runtime APIs. You're going to need to use mkbundle in a slightly different workflow so you can make a file that it needs to be passed. 

If you have any problems working out the steps, notify me here or on Gitter and I will try to create a better guide. I am not familiar with the user-friendliness of 170e9442c976ed6cd4a499093e11a9e12e7368bb

**Comment by alexanderkyte** [comment]
https://github.com/mono/mono/pull/10170

**Comment by parkerlreed** [comment]
Running into this when trying to build a project found on Github

Arch Linux
````
gcc 8.2.1+20180831-1
mono 5.16.0.179-1
mono-addins 1.3.3-2
````
````
[parker@e55 build]$ git clone --recurse-submodules  https://github.com/jealouscloud/linerider-advanced
[parker@e55 build]$ cd linerider-advanced/src
[parker@e55 src]$ nuget restore
[parker@e55 src]$ xbuild 
[parker@e55 src]$ cd ../build/
[parker@e55 build]$ mkbundle -L /usr/lib/mono/4.6.2-api/ --deps linerider.exe -o linerider
ERROR: Unable to load assembly `System.Runtime' referenced by `/home/parker/build/linerider-advanced/build/System.Threading.Tasks.Extensions.dll'
[parker@e55 build]$ sudo cp /usr/lib/mono/4.6.2-api/Facades/System.* /usr/lib/mono/4.6.2-api/
[parker@e55 build]$ mkbundle -L /usr/lib/mono/4.6.2-api/ --deps linerider.exe -o linerider
temp.c: In function ‘init_default_mono_api_struct’:
temp.c:192:33: warning: assignment to ‘void (*)(int)’ from incompatible pointer type ‘void (*)(MonoAotMode)’ {aka ‘void (*)(enum <anonymous>)’} [-Wincompatible-pointer-types]
  mono_api.mono_jit_set_aot_mode = mono_jit_set_aot_mode;
                                 ^
temp.c:193:38: error: ‘mono_aot_register_module’ undeclared (first use in this function); did you mean ‘mono_jit_set_aot_mode’?
  mono_api.mono_aot_register_module = mono_aot_register_module;
                                      ^~~~~~~~~~~~~~~~~~~~~~~~
                                      mono_jit_set_aot_mode
temp.c:193:38: note: each undeclared identifier is reported only once for each function it appears in
temp.c: In function ‘install_dll_config_files’:
temp.c:209:60: warning: pointer targets in passing argument 2 of ‘mono_api.mono_register_config_for_assembly’ differ in signedness [-Wpointer-sign]
  mono_api.mono_register_config_for_assembly ("OpenTK.dll", assembly_config_OpenTK_dll);
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
temp.c:209:60: note: expected ‘const char *’ but argument is of type ‘const unsigned char *’
At top level:
temp.c:187:1: warning: ‘init_default_mono_api_struct’ defined but not used [-Wunused-function]
 init_default_mono_api_struct ()
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
temp.c:168:1: warning: ‘validate_api_struct’ defined but not used [-Wunused-function]
 validate_api_struct ()
 ^~~~~~~~~~~~~~~~~~~
ERROR: [Fail] 256
````

If I'm just doing something horribly wrong, please let me know.

**Comment by Ronaldho80** [comment]
I still have this issue with the newest Ubuntu version of Mono (5.16.0.179)

## Steps to Reproduce

    This part of my Makefile:
```
# all dependent assemblies                                                                                                         
 15 #                                                                                                                                  
 14 SOURCES=Zlib.cs GZipCompressor.cs SGML.cs Program.cs                                                                               
 13                                                                                                                                    
 12 DEPS=packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll \                                              
 11      packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.RawFileReader.dll \                                     
 10      packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.MassPrecisionEstimator.dll \                            
  9      packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll \                                              
  8      packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.BackgroundSubtraction.dll \                             
  7      packages/NDesk.Options.0.2.1/lib/NDesk.Options.dll \                                                                          
  6      packages/Ionic.Zip.1.9.1.8/lib/Ionic.Zip.dll                                                                                  
  5                                                                                                                                    
  4 # add '-r:' option to each assembly                                                                                                
  3 PACKAGES=$(addprefix -r:,$(DEPS))                                                                                                  
  2                                                                                                                                    
  1 bundle: compile                                                                                                                    
  0     mkbundle --nodeps --static -o RawConverter2 RawConvert2.exe -L /usr/lib/mono/4.5/ $(DEPS)                                      
```
## Current Behavior

Error:
```
herzog@Elfriede:~/Lipotype/RawConverter2 
$ make
mcs -o -debug- -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.RawFileReader.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.MassPrecisionEstimator.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.BackgroundSubtraction.dll -r:packages/NDesk.Options.0.2.1/lib/NDesk.Options.dll -r:packages/Ionic.Zip.1.9.1.8/lib/Ionic.Zip.dll Zlib.cs GZipCompressor.cs SGML.cs Program.cs -out:RawConvert2.exe
warning CS8029: Compatibility: Use -out:FILE instead of --output FILE or -o FILE
mkbundle --nodeps --static -o RawConverter2 RawConvert2.exe -L /usr/lib/mono/4.5/ packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.RawFileReader.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.MassPrecisionEstimator.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.BackgroundSubtraction.dll packages/NDesk.Options.0.2.1/lib/NDesk.Options.dll packages/Ionic.Zip.1.9.1.8/lib/Ionic.Zip.dll
temp.c: In function ‘init_default_mono_api_struct’:
temp.c:134:33: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
  mono_api.mono_jit_set_aot_mode = mono_jit_set_aot_mode;
                                 ^
temp.c:135:38: error: ‘mono_aot_register_module’ undeclared (first use in this function)
  mono_api.mono_aot_register_module = mono_aot_register_module;
                                      ^
temp.c:135:38: note: each undeclared identifier is reported only once for each function it appears in
temp.c: At top level:
temp.c:110:1: warning: ‘validate_api_struct’ defined but not used [-Wunused-function]
 validate_api_struct ()
 ^
temp.c:129:1: warning: ‘init_default_mono_api_struct’ defined but not used [-Wunused-function]
 init_default_mono_api_struct ()
 ^
ERROR: [Fail] 256
Makefile:17: recipe for target 'bundle' failed
make: *** [bundle] Error 1

Expected Behavior

It compiles a bundle
On which platforms did you notice this

[ ] macOS
[ x] Linux
[ ] Windows
```
## Version Used:
```
herzog@Elfriede:~/Lipotype/RawConverter2 
$ mono --version
Mono JIT compiler version 5.16.0.179 (tarball Thu Oct  4 10:29:10 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(3.6.0svn-mono-/)
	GC:            sgen (concurrent by default)
```

**Comment by batmaninpink** [comment]
Isn't this already fixed with https://github.com/mono/mono/pull/10170 ?

**Comment by Herz3h** [comment]
I have same problem on Ubutnu 18.04 when I type following command : 

`mkbundle -o MergeHeaderAndDocument MergeHeaderAndDocument.exe -L /usr/lib/mono/4.7-api
`
My version of mono : 

```
Mono JIT compiler version 5.16.0.220 (tarball Mon Nov 26 17:17:59 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(3.6.0svn-mono-/)
	GC:            sgen (concurrent by default)

```

**Comment by parkerlreed** [comment]
@batmaninpink Thanks! I'll give that a shot later.

**Comment by Ronaldho80** [comment]
I just upgraded my mono installation on Ubuntu to 5.18.0.240:
```
$ mono --version
Mono JIT compiler version 5.18.0.240 (tarball Wed Jan 16 09:13:03 UTC 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(600)
	Suspend:       preemptive
	GC:            sgen (concurrent by default)
```
However, contrary to @batmaninpink suggestion, it still produces the same error:
```
$ make
mcs -o -debug- -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.RawFileReader.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.MassPrecisionEstimator.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll -r:packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.BackgroundSubtraction.dll -r:packages/NDesk.Options.0.2.1/lib/NDesk.Options.dll -r:packages/Ionic.Zip.1.9.1.8/lib/Ionic.Zip.dll Zlib.cs GZipCompressor.cs SGML.cs Program.cs -out:RawConvert2.exe
warning CS8029: Compatibility: Use -out:FILE instead of --output FILE or -o FILE
mkbundle --nodeps --static -o RawConverter2 RawConvert2.exe -L /usr/lib/mono/4.7-api/ packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.RawFileReader.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.MassPrecisionEstimator.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.Data.dll packages/ThermoRawFileReader.3.0.41/Libraries/ThermoFisher.CommonCore.BackgroundSubtraction.dll packages/NDesk.Options.0.2.1/lib/NDesk.Options.dll packages/Ionic.Zip.1.9.1.8/lib/Ionic.Zip.dll
temp.c: In function ‘init_default_mono_api_struct’:
temp.c:134:33: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
  mono_api.mono_jit_set_aot_mode = mono_jit_set_aot_mode;
                                 ^
temp.c:135:38: error: ‘mono_aot_register_module’ undeclared (first use in this function)
  mono_api.mono_aot_register_module = mono_aot_register_module;
                                      ^
temp.c:135:38: note: each undeclared identifier is reported only once for each function it appears in
temp.c: At top level:
temp.c:110:1: warning: ‘validate_api_struct’ defined but not used [-Wunused-function]
 validate_api_struct ()
 ^
temp.c:129:1: warning: ‘init_default_mono_api_struct’ defined but not used [-Wunused-function]
 init_default_mono_api_struct ()
 ^
ERROR: [Fail] 256
Makefile:17: recipe for target 'bundle' failed
make: *** [bundle] Error 1

```

Is there another way to make the bundle instead of using mkbundle?

Thx, Ronny

**Comment by petlof** [comment]
@Ronaldho80 Add --simple as argument to mkbundle and it will work!

**Comment by Ronaldho80** [comment]
Thx a lot, this did it!

**Comment by njskalski** [comment]
ha ha!
»ʚ mkbundle --simple Freeserf.net/bin/Release/Freeserf.net.exe --deps -o freeserf.net -L /usr/lib/mono/4.6.1-api -L Freeserf.Core/release -L /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib -L Freeserf.Renderer.OpenTK/release -L packages/OpenTK.GLControl.3.0.1/lib/net20/ -L Freeserf.Network/release/ --machine-config /etc/mono/4.5/machine.config --config /etc/mono/4.5/machine.config 
Using runtime: /usr/bin/mono
     Assembly: /home/andrzej/r/freeserf.net/Freeserf.net/bin/Release/Freeserf.net.exe
       Config: /home/andrzej/r/freeserf.net/Freeserf.net/bin/Release/Freeserf.net.exe.config
     Assembly: /usr/lib/mono/4.6.1-api/mscorlib.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Windows.Forms.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Configuration.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Xml.dll
     Assembly: /usr/lib/mono/4.6.1-api/Accessibility.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Drawing.dll
     Assembly: /home/andrzej/r/freeserf.net/Freeserf.Core/release/Freeserf.Core.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/netstandard.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Core.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Data.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.EnterpriseServices.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Transactions.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Data.Common.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Diagnostics.StackTrace.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Diagnostics.Tracing.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Globalization.Extensions.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.IO.Compression.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.IO.Compression.FileSystem.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.ComponentModel.Composition.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Net.Http.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Net.Sockets.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Numerics.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.ValueTuple.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Runtime.InteropServices.RuntimeInformation.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Runtime.Serialization.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Runtime.Serialization.Xml.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Runtime.Serialization.Primitives.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Security.Cryptography.Algorithms.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Security.SecureString.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Threading.Overlapped.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Web.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Web.ApplicationServices.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.ComponentModel.DataAnnotations.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Web.Services.dll
     Assembly: /usr/lib/mono/4.6.1-api/System.Xml.Linq.dll
     Assembly: /usr/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Xml.XPath.XDocument.dll
     Assembly: /home/andrzej/r/freeserf.net/Freeserf.Core/release/System.Reflection.Emit.Lightweight.dll
     Assembly: /home/andrzej/r/freeserf.net/Freeserf.Core/release/System.Reflection.Emit.ILGeneration.dll
     Assembly: /usr/lib/mono/4.6.1-api/Microsoft.CSharp.dll
     Assembly: /home/andrzej/r/freeserf.net/Freeserf.Renderer.OpenTK/release/Freeserf.Renderer.OpenTK.dll
     Assembly: /home/andrzej/r/freeserf.net/Freeserf.Renderer.OpenTK/release/OpenTK.dll
     Assembly: /home/andrzej/r/freeserf.net/packages/OpenTK.GLControl.3.0.1/lib/net20/OpenTK.GLControl.dll
     Assembly: /home/andrzej/r/freeserf.net/Freeserf.Network/release/Freeserf.Network.dll
     Assembly: /usr/lib/mono/4.6.1-api/I18N.West.dll
     Assembly: /usr/lib/mono/4.6.1-api/I18N.dll
systemconfig: /etc/mono/4.5/machine.config
machineconfig: /etc/mono/4.5/machine.config

[15:53:11] andrzej@mars /home/andrzej/r/freeserf.net (e: 0, t: 13) 
»ʚ ./freeserf.net
The file /usr/lib/mscorlib.dll is an invalid CIL image

**Comment by alexanderkyte** [comment]
Closing this as the fix here: https://github.com/mono/mono/pull/10170 is expected to have fixed it. You may be using older Monos that don't contain that fix. Please reopen issue if you can confirm it's still broken with a newer Mono.

**Comment by alexanderkyte** [comment]
@Ronaldho80 you may be running into related issues involving the mkbundle API change. If you still see them, could you file a separate issue to be investigated?
